src/main/scala/org/wikimedia/analytics/hdfstools/HdfsRsyncConfig.scala

```diff
@@ -249,6 +306,81 @@ case class HdfsRsyncConfig(
         prepareErrorMessage(errorMessages, "Error validating chmod commands:")
     }
 
+    def validateChown: Option[String] = {
+        val errorMessages = chown.map(chownValue => {
+            val parsingErrors = {
+                val parts = chownValue.split(":")
+                val partsNumber = parts.length
+                if (partsNumber == 1 || partsNumber == 2) {
+                    val username = parts(0)
+                    val usernameErrors = {
+                        if (username.length > 0) {
+                            (if (mappingNamePattern.findFirstIn(username).isEmpty) Seq(s"Invalid username: $username") else Nil) ++
+                                (if (username.length > maxNameSize) Seq(s"Invalid username size (max 32): $username") else Nil)
+                        } else {
+                            if (parts.length == 1) Seq(s"Invalid empty username") else Nil
+                        }
+                    }
+
+                    val groupnameErrors = {
+                        if (partsNumber == 2) {
+                            val groupname = parts(1)
+                            if (groupname.length > 0) {
+                                (if (mappingNamePattern.findFirstIn(groupname).isEmpty) Seq(s"Invalid groupname: $groupname") else Nil) ++
+                                    (if (groupname.length > maxNameSize) Seq(s"Invalid groupname size (max 32): $groupname") else Nil)
+                            } else {
+                                // No need to check for empty username as ":".split(":") returns emtpy array
+                                Nil
+                            }
+                        } else {
+                            Nil
+                        }
+                    }
+
+                    usernameErrors ++ groupnameErrors
+
+                } else {
+                    Seq(s"Invalid chown format: $chownValue")
+                }
+            }
+
+            (if (usermap.nonEmpty || groupmap.nonEmpty) Seq("chown and usermap/groupmap cannot be used simultaneously") else Nil) ++
+                parsingErrors
+
+        }).getOrElse(Seq.empty)
+
+        prepareErrorMessage(errorMessages, s"Error validating chown:")
+    }
+
+    /**
+     * Function validating a list of user or group mappings.
+     * Mappings should be in the format: 'pattern:value' where pattern will be matched against
```

Comment:
This doc might be useful in the CLI opt help messages too.

Reply:
Done